Gems/ScriptCanvas/Code/Include/ScriptCanvas/Execution/ExecutionState.cpp

```diff
@@ -6,118 +6,132 @@
  *
  */
 
-#include <AzCore/RTTI/BehaviorContext.h>
-#include <ScriptCanvas/Core/Core.h>
+#include <AzCore/Asset/AssetManager.h>
+#include <ScriptCanvas/Asset/RuntimeAsset.h>
 #include <ScriptCanvas/Execution/RuntimeComponent.h>
 
-#include "Interpreted/ExecutionStateInterpreted.h"
-#include "Interpreted/ExecutionStateInterpretedPure.h"
-#include "Interpreted/ExecutionStateInterpretedPerActivation.h"
-#include "Interpreted/ExecutionStateInterpretedSingleton.h"
+#include <ScriptCanvas/Execution/ExecutionState.h>
 
-#include "ExecutionState.h"
+namespace ExecutionStateCpp
+{
+    using namespace ScriptCanvas;
+
+    AZ::Data::Asset<RuntimeAsset> GetSubgraphAssetForDebug(const AZ::Data::AssetId& id)
+    {
+        // #functions2 this may have to be made recursive
+        auto asset = AZ::Data::AssetManager::Instance().GetAsset<SubgraphInterfaceAsset>(id, AZ::Data::AssetLoadBehavior::PreLoad);
+        asset.BlockUntilLoadComplete();
+        return asset;
+    }
+}
 
 namespace ScriptCanvas
 {
-    ExecutionStateConfig::ExecutionStateConfig(AZ::Data::Asset<RuntimeAsset> runtimeAsset, RuntimeComponent& component)
-        : asset(runtimeAsset)
-        , component(component)
-        , runtimeData(runtimeAsset.Get()->m_runtimeData)
+    ExecutionStateConfig::ExecutionStateConfig(const RuntimeDataOverrides& overrides, ExecutionUserData&& userData)
+        : runtimeData(overrides.m_runtimeAsset.Get()->m_runtimeData)
+        , overrides(overrides)
+        , userData(AZStd::move(userData))
     {}
-
-    ExecutionState::ExecutionState(const ExecutionStateConfig& config)
-        : m_component(&config.component)
+    
+    ExecutionState::ExecutionState(ExecutionStateConfig& config)
+        : m_runtimeData(config.runtimeData)
+        , m_overrides(config.overrides)
+        , m_userData(AZStd::move(config.userData))
     {}
 
-    ExecutionStatePtr ExecutionState::Create(const ExecutionStateConfig& config)
+    AZ::Data::AssetId ExecutionState::GetAssetId() const
     {
-        Grammar::ExecutionStateSelection selection = config.runtimeData.m_input.m_executionSelection;
-
-        switch (selection)
-        {
-        case Grammar::ExecutionStateSelection::InterpretedPure:
-            return AZStd::make_shared<ExecutionStateInterpretedPure>(config);
-
-        case Grammar::ExecutionStateSelection::InterpretedPureOnGraphStart:
-            return AZStd::make_shared<ExecutionStateInterpretedPureOnGraphStart>(config);
-
-        case Grammar::ExecutionStateSelection::InterpretedObject:
-            return AZStd::make_shared<ExecutionStateInterpretedPerActivation>(config);
+        return m_overrides.m_runtimeAsset.GetId();
```

Comment:
what is the difference between override and runtimeData? why not get id from runtimeData?

Reply:
I'm going to to a separate those and clean up the names, but runtimeData are the immutable guts of the asset, and the overrides are the properties that are set and could override the default values in the runtimeData, for example when prefab values are overridden. Getting the Id from the runtime data is a good callout though, and I can switch that.